### PR TITLE
Clarity: additional token functions

### PR DESCRIPTION
**Pull request by lgalabru**
Address https://github.com/blockstack/stacks-blockchain/issues/2125, by implementing `ft-get-supply`, `ft-burn?`, `nft-burn?`.
Theoretically code complete, still need to:

- [x] Add tests
- [x] Fill cost contract

### CODE REVIEW

**Comment by psq** [comment]
Re these going to require new post conditions?

**Review by jcnelson — APPROVED** [review]
LGTM!  Great work @lgalabru!

File: src/vm/docs/mod.rs [hunk 1]
```diff
@@ -1449,6 +1449,58 @@ one of the following error codes:
...
+const GET_TOKEN_SUPPLY: SpecialAPI = SpecialAPI {
+    input_type: "TokenName",
+    output_type: "uint",
+    signature: "(ft-get-supply token-name)",
+    description: "`ft-get-balance` returns `token-name` circulating supply.
+The token type must have been defined using `define-fungible-token`.",
+    example: "
+(define-fungible-token stackaroo)
+(ft-mint? stackaroo u100 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)
+(ft-get-supply stackaroo) ;; returns u100
```
Comment (kantai) [inline src/vm/docs/mod.rs:1461]: Can you capitalize `Returns`? That way, the `docs::test::test_examples` test will check that the output matches expectation (or change the docs test to be case-insensitive in this detection).

Comment (lgalabru) [reply src/vm/docs/mod.rs]: Addressed with https://github.com/blockstack/stacks-blockchain/pull/2164/commits/44f7ade78ffff88cf2f23dadc8558ce8b7e0a642

File: src/chainstate/stacks/events.rs [hunk 2]
```diff
@@ -279,6 +295,30 @@ impl NFTMintEventData {
...
+    pub fn json_serialize(&self) -> serde_json::Value {
+        let raw_value = {
+            let mut bytes = vec![];
+            self.value.consensus_serialize(&mut bytes).unwrap();
+            let formatted_bytes: Vec<String> = bytes.iter().map(|b| format!("{:02x}", b)).collect();
+            formatted_bytes
+        };
+        json!({
+            "asset_identifier": format!("{}", self.asset_identifier),
+            "sender": format!("{}",self.sender),
```
Comment (kantai) [inline src/chainstate/stacks/events.rs:315]: No need to change, but just a note that `.to_string()` is a an auto-implemented function that is more or less identical to `format!("{}", ...)`.

File: testnet/stacks-node/src/tests/neon_integrations.rs [hunk 3]
```diff
@@ -1987,57 +1987,151 @@ fn atlas_integration_test() {
...
-
-        // The bootstrap node mined a few blocks and is ready, let's setup this node.
-        match follower_node_rx.recv() {
-            Ok(Signal::BootstrapNodeReady) => {
-                println!("Booting follower node...");
-            }
-            _ => panic!("Bootstrap node could nod boot. Aborting test."),
-        };
+    // Start the attached observer
+    test_observer::spawn();
```
Comment (kantai) [inline testnet/stacks-node/src/tests/neon_integrations.rs:1991]: Did the follower_node not need to be spawned in a different thread?

Comment (lgalabru) [reply testnet/stacks-node/src/tests/neon_integrations.rs:1991]: we only need to have one of the node on a separate thread.  I've been observing some flakiness between the integration tests, with the test_observer not being cleared (previously done in a separate thread, now on the main thread) after the atlas test. This refactoring in an attempt to fix this issue.

**Review by kantai — APPROVED** [review]
This LGTM, I just had a question about the neon_integrations change (not a blocking comment).

**Pull request merged**